cpp/shared/TestResult.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/* 
+ * File:   TestResult.h
+ * Author: mccaffrey
+ *
+ * Created on May 20, 2020, 7:10 PM
+ */
+
+#ifndef TESTRESULT_H
+#define TESTRESULT_H
+
+#include <cstdlib>
+#include <string>
+
+
+
+using namespace std;
+
+class TestResult {
+private:
```

Comment:
There are a bunch of functions that intermediate between a nominally internal data set the stored value, but do absolutely nothing else. Why is this helpful? If one has the ability to change the value to anything with no controls or checks, why not make the data items public and remove the indirection?